code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -288,3 +294,9 @@ Pipelines + Other Objects -> Pipe network
 //Used for certain children of obj/machinery/atmospherics to not show pipe vision when mob is inside it.
 /obj/machinery/atmospherics/proc/can_see_pipes()
 	return 1
+
+/obj/machinery/atmospherics/fire_act(exposed_temperature, exposed_volume)
+	var/turf/T = src.loc
+	if(T && T.intact && level == 1) //protected from fire when hidden behind a floor.
+		return

```

Comment:
This trick protects even scrubbers which can be seen even when the turf is intact. So the scrubber and vent in burn chambers don't melt.